ironfish/src/assert.ts

```diff
@@ -7,6 +7,10 @@ export class Assert {
     throw new Error(x)
   }
 
+  static isUndefined<T>(x: undefined | T, message?: string): asserts x is undefined {
+    if (x !== undefined) throw new Error(message || `Expected ${String(x)} to be undefined`)
```

Comment:
It would be nice to use `JSON.stringify` here so we get better logging for objects.